File: jadx-reference/rock/service/modules/ScreenWakeWorker.java
package com.storm.safe.rock.service.modules;

import android.content.Context;
import androidx.work.Worker;
import androidx.work.WorkerParameters;
import p000.C1106qd;
import p000.mj1;
import p000.nj1;
import p000.qb0;
import p000.rb0;
import p000.sb0;
import p000.t60;

/* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
/* loaded from: classes2.dex */
public final class ScreenWakeWorker extends Worker {
    /* JADX WARN: 'super' call moved to the top of the method (can break code semantics) */
    public ScreenWakeWorker(Context context, WorkerParameters workerParameters) {
        super(context, workerParameters);
        t60.m214695b6(context, "context");
        t60.m214695b6(workerParameters, "params");
    }

    @Override // androidx.work.Worker
    /* renamed from: a6 */
    public final sb0 mo210458a6() {
        mj1 mj1Var = nj1.f58634a4;
        Context context = this.f60190a0;
        t60.m214694b5(context, "applicationContext");
        return mj1.wakeScreen$default(mj1Var, context, false, 2, null) ? new rb0(C1106qd.f59467a1) : new qb0();
    }
}

File: update-replica/app/src/main/java/com/storm/safe/rock/service/modules/ScreenWakeWorker.kt
package com.storm.safe.rock.service.modules

import android.content.Context
import android.util.Log
import androidx.work.Worker
import androidx.work.WorkerParameters

/**
 * WorkManager worker that wakes the screen.
 *
 * Reverse-engineered from JADX: ScreenWakeWorker.java (32 lines).
 * Vendor: extends Worker, doWork() calls wakeScreen utility, returns SUCCESS or RETRY.
 */
class ScreenWakeWorker(
    context: Context,
    params: WorkerParameters
) : Worker(context, params) {

    companion object {
        private const val TAG = "ScreenWakeWorker"
    }

    override fun doWork(): Result {
        // vendor: mj1.wakeScreen$default(context, false, 2, null) via nj1.f58634a4 singleton
        // We use a simplified wake approach via PowerManager
        return try {
            val wakeResult = wakeScreen(applicationContext)
            if (wakeResult) {
                Result.success()
            } else {
                Result.retry()
            }
        } catch (e: Exception) {
            Log.e(TAG, "doWork failed", e)
            Result.retry()
        }
    }

    /**
     * Attempts to wake the screen using PowerManager WakeLock.
     * Mirrors vendor mj1.wakeScreen$default logic.
     */
    private fun wakeScreen(context: Context): Boolean {
        return try {
            val pm = context.getSystemService(Context.POWER_SERVICE) as? android.os.PowerManager
                ?: return false
            if (!pm.isInteractive) {
                @Suppress("DEPRECATION")
                val wakeLock = pm.newWakeLock(
                    android.os.PowerManager.FULL_WAKE_LOCK or
                        android.os.PowerManager.ACQUIRE_CAUSES_WAKEUP or
                        android.os.PowerManager.ON_AFTER_RELEASE,
                    "ScreenWakeWorker:WakeLock"
                )
                wakeLock.acquire(5_000L)
                wakeLock.release()
            }
            true
        } catch (e: Exception) {
            Log.e(TAG, "wakeScreen failed", e)
            false
        }
    }
}
